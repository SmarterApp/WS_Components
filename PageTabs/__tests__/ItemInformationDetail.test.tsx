import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ItemInformationDetail } from '../ItemInformationDetail';
import { shallow, mount } from 'enzyme';

describe("ItemInformationDetail", () => {
    const props = {
        itemCardViewModel: {
            bankKey: 1,
            itemKey: 123,
            title: "test_title",
            grade: 1,
            gradeLabel: "1",
            subjectCode: "Math",
            subjectLabel: "MTH",
            claimCode: "claim_1234",
            claimLabel: "claim_math",
            target: "",
            interactionTypeCode: "type01",
            interactionTypeLabel: "type01label",
            isPerformanceItem: false
        },
        depthOfKnowledge: "",
        targetDescription: "",
        commonCoreStandardsDescription: "",
        educationalDifficulty: "",
        evidenceStatement: ""
    }

    it("matches snapshot", () => {
        expect(shallow(<ItemInformationDetail {...props} />)).toMatchSnapshot();
    })
})