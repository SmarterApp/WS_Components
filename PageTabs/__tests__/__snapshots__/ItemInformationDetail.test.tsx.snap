// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemInformationDetail matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemInformationDetail
    commonCoreStandardsDescription=""
    depthOfKnowledge=""
    educationalDifficulty=""
    evidenceStatement=""
    itemCardViewModel={
      Object {
        "bankKey": 1,
        "claimCode": "claim_1234",
        "claimLabel": "claim_math",
        "grade": 1,
        "gradeLabel": "1",
        "interactionTypeCode": "type01",
        "interactionTypeLabel": "type01label",
        "isPerformanceItem": false,
        "itemKey": 123,
        "subjectCode": "Math",
        "subjectLabel": "MTH",
        "target": "",
        "title": "test_title",
      }
    }
    targetDescription=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="item-information-content"
      >
        <p
          className="card-text subject"
          tabIndex={0}
        >
          <span
            className="card-text-label"
          >
            Subject
          </span>
          <span
            className="card-text-value"
          >
             
            MTH
          </span>
        </p>
        <p
          className="card-text grade"
          tabIndex={0}
        >
          <span
            className="card-text-label"
          >
            Grade
          </span>
          <span
            className="card-text-value"
          >
             
            1
          </span>
        </p>
        <p
          className="card-text claim"
          tabIndex={0}
        >
          <span
            className="card-text-label"
          >
            Claim
          </span>
          <span
            className="card-text-value"
          >
             
            claim_math
          </span>
        </p>
        <p
          className="card-text interaction-type"
          tabIndex={0}
        >
          <span
            className="card-text-label"
          >
            Item Type
          </span>
          <span
            className="card-text-value"
          >
             
            type01label
          </span>
        </p>
        <p
          className="card-text item-id"
          tabIndex={0}
        >
          <span
            className="card-text-label"
          >
            Item Id
          </span>
          <span
            className="card-text-value"
          >
             
            123
          </span>
        </p>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="card-text subject"
            tabIndex={0}
          >
            <span
              className="card-text-label"
            >
              Subject
            </span>
            <span
              className="card-text-value"
            >
               
              MTH
            </span>
          </p>,
          <p
            className="card-text grade"
            tabIndex={0}
          >
            <span
              className="card-text-label"
            >
              Grade
            </span>
            <span
              className="card-text-value"
            >
               
              1
            </span>
          </p>,
          <p
            className="card-text claim"
            tabIndex={0}
          >
            <span
              className="card-text-label"
            >
              Claim
            </span>
            <span
              className="card-text-value"
            >
               
              claim_math
            </span>
          </p>,
          null,
          <p
            className="card-text interaction-type"
            tabIndex={0}
          >
            <span
              className="card-text-label"
            >
              Item Type
            </span>
            <span
              className="card-text-value"
            >
               
              type01label
            </span>
          </p>,
          <p
            className="card-text item-id"
            tabIndex={0}
          >
            <span
              className="card-text-label"
            >
              Item Id
            </span>
            <span
              className="card-text-value"
            >
               
              123
            </span>
          </p>,
          null,
          null,
          null,
          null,
          null,
        ],
        "className": "item-information-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-text-label"
              >
                Subject
              </span>,
              <span
                className="card-text-value"
              >
                 
                MTH
              </span>,
            ],
            "className": "card-text subject",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Subject",
                "className": "card-text-label",
              },
              "ref": null,
              "rendered": "Subject",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "MTH",
                ],
                "className": "card-text-value",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "MTH",
              ],
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-text-label"
              >
                Grade
              </span>,
              <span
                className="card-text-value"
              >
                 
                1
              </span>,
            ],
            "className": "card-text grade",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Grade",
                "className": "card-text-label",
              },
              "ref": null,
              "rendered": "Grade",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "1",
                ],
                "className": "card-text-value",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "1",
              ],
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-text-label"
              >
                Claim
              </span>,
              <span
                className="card-text-value"
              >
                 
                claim_math
              </span>,
            ],
            "className": "card-text claim",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Claim",
                "className": "card-text-label",
              },
              "ref": null,
              "rendered": "Claim",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "claim_math",
                ],
                "className": "card-text-value",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "claim_math",
              ],
              "type": "span",
            },
          ],
          "type": "p",
        },
        null,
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-text-label"
              >
                Item Type
              </span>,
              <span
                className="card-text-value"
              >
                 
                type01label
              </span>,
            ],
            "className": "card-text interaction-type",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Item Type",
                "className": "card-text-label",
              },
              "ref": null,
              "rendered": "Item Type",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "type01label",
                ],
                "className": "card-text-value",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "type01label",
              ],
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-text-label"
              >
                Item Id
              </span>,
              <span
                className="card-text-value"
              >
                 
                123
              </span>,
            ],
            "className": "card-text item-id",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Item Id",
                "className": "card-text-label",
              },
              "ref": null,
              "rendered": "Item Id",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  123,
                ],
                "className": "card-text-value",
              },
              "ref": null,
              "rendered": Array [
                " ",
                123,
              ],
              "type": "span",
            },
          ],
          "type": "p",
        },
        null,
        null,
        null,
        null,
        null,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="item-information-content"
        >
          <p
            className="card-text subject"
            tabIndex={0}
          >
            <span
              className="card-text-label"
            >
              Subject
            </span>
            <span
              className="card-text-value"
            >
               
              MTH
            </span>
          </p>
          <p
            className="card-text grade"
            tabIndex={0}
          >
            <span
              className="card-text-label"
            >
              Grade
            </span>
            <span
              className="card-text-value"
            >
               
              1
            </span>
          </p>
          <p
            className="card-text claim"
            tabIndex={0}
          >
            <span
              className="card-text-label"
            >
              Claim
            </span>
            <span
              className="card-text-value"
            >
               
              claim_math
            </span>
          </p>
          <p
            className="card-text interaction-type"
            tabIndex={0}
          >
            <span
              className="card-text-label"
            >
              Item Type
            </span>
            <span
              className="card-text-value"
            >
               
              type01label
            </span>
          </p>
          <p
            className="card-text item-id"
            tabIndex={0}
          >
            <span
              className="card-text-label"
            >
              Item Id
            </span>
            <span
              className="card-text-value"
            >
               
              123
            </span>
          </p>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="card-text subject"
              tabIndex={0}
            >
              <span
                className="card-text-label"
              >
                Subject
              </span>
              <span
                className="card-text-value"
              >
                 
                MTH
              </span>
            </p>,
            <p
              className="card-text grade"
              tabIndex={0}
            >
              <span
                className="card-text-label"
              >
                Grade
              </span>
              <span
                className="card-text-value"
              >
                 
                1
              </span>
            </p>,
            <p
              className="card-text claim"
              tabIndex={0}
            >
              <span
                className="card-text-label"
              >
                Claim
              </span>
              <span
                className="card-text-value"
              >
                 
                claim_math
              </span>
            </p>,
            null,
            <p
              className="card-text interaction-type"
              tabIndex={0}
            >
              <span
                className="card-text-label"
              >
                Item Type
              </span>
              <span
                className="card-text-value"
              >
                 
                type01label
              </span>
            </p>,
            <p
              className="card-text item-id"
              tabIndex={0}
            >
              <span
                className="card-text-label"
              >
                Item Id
              </span>
              <span
                className="card-text-value"
              >
                 
                123
              </span>
            </p>,
            null,
            null,
            null,
            null,
            null,
          ],
          "className": "item-information-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="card-text-label"
                >
                  Subject
                </span>,
                <span
                  className="card-text-value"
                >
                   
                  MTH
                </span>,
              ],
              "className": "card-text subject",
              "tabIndex": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Subject",
                  "className": "card-text-label",
                },
                "ref": null,
                "rendered": "Subject",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "MTH",
                  ],
                  "className": "card-text-value",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "MTH",
                ],
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="card-text-label"
                >
                  Grade
                </span>,
                <span
                  className="card-text-value"
                >
                   
                  1
                </span>,
              ],
              "className": "card-text grade",
              "tabIndex": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Grade",
                  "className": "card-text-label",
                },
                "ref": null,
                "rendered": "Grade",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "1",
                  ],
                  "className": "card-text-value",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "1",
                ],
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="card-text-label"
                >
                  Claim
                </span>,
                <span
                  className="card-text-value"
                >
                   
                  claim_math
                </span>,
              ],
              "className": "card-text claim",
              "tabIndex": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Claim",
                  "className": "card-text-label",
                },
                "ref": null,
                "rendered": "Claim",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "claim_math",
                  ],
                  "className": "card-text-value",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "claim_math",
                ],
                "type": "span",
              },
            ],
            "type": "p",
          },
          null,
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="card-text-label"
                >
                  Item Type
                </span>,
                <span
                  className="card-text-value"
                >
                   
                  type01label
                </span>,
              ],
              "className": "card-text interaction-type",
              "tabIndex": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Item Type",
                  "className": "card-text-label",
                },
                "ref": null,
                "rendered": "Item Type",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "type01label",
                  ],
                  "className": "card-text-value",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "type01label",
                ],
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="card-text-label"
                >
                  Item Id
                </span>,
                <span
                  className="card-text-value"
                >
                   
                  123
                </span>,
              ],
              "className": "card-text item-id",
              "tabIndex": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Item Id",
                  "className": "card-text-label",
                },
                "ref": null,
                "rendered": "Item Id",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    123,
                  ],
                  "className": "card-text-value",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  123,
                ],
                "type": "span",
              },
            ],
            "type": "p",
          },
          null,
          null,
          null,
          null,
          null,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
